src/main/java/org/apache/commons/dbcp2/PoolableConnectionFactory.java

```diff
@@ -94,6 +100,8 @@
 
     private Integer defaultQueryTimeoutSeconds;
 
+    private final ExecutorService executor = Executors.newFixedThreadPool(3, new AbortThreadFactory());
```

Comment:
Should we provide an override through a system property?

Reply:
If it needs to be configured why a system property rather than a property of the factory?

Reply:
Not sure what you mean exactly, here.  I would be +0 to add two configuration properties:  abortOnAbandoned (to turn this feature on - default to true) and abandonedThreadPoolSize (or better name) defaulting to 3.

Reply:
Why not making it synchronous? (Executor e = Runnable::run)
In this state it can fail at instantiation if system already uses a lot of threads (and leak), leak at shutdown/undeploy time (an issue for OSGi and application servers) due to daemon thread usage and the lack of await at close time.
If really desired to be thread based, minimum for me is to 1. await the stop properly (= handle the lifecycle completely and not assume it will be done fast after close() is called since quickly after close the classloader can be destroyed and finalization can fail), 2. make the executor lazy (if no connection are abandonned it is pointless to consume these resources and to slow down the instantiation/startup)

That said I'm really to make it synchronous since the async there just brings issues IMHO.

Reply:
> If it needs to be configured why a system property rather than a property of the factory?

Sure, I just wanted to avoid not having an override. I meant the sys prop to be a placeholder hack as an example impl.

Reply:
I was also worried about the issues @rmannibucau mentions, but did not think of the option he recommends.  Unless I am missing something, I think it should work and would be +1 to that change.  Thanks for the review!

Reply:
Wanting to keep that ball rolling on this PR...

>I think it should work and would be +1 to that change. 

@psteitz : Are you planning on doing this in this PR or are you waiting for the PR to be merged?

Reply:
ping @psteitz

Reply:
Hi @rmannibucau 
I merged this PR and implemented my 2 comments.
May you please follow up on your comment above with a new PR?
TY!